### PR TITLE
Prevent Menu From Closing When Turning On or Off Sensors to Display

### PR DESCRIPTION
**Pull request by xmikedanielsx**
Request Person: JayzTwoCents
Request From: https://www.youtube.com/watch?v=sX2ThUWNXio
Request:
As an end User I have a problem with when selecting sensors to show or hide in the Hardware menu, the menu closes. This is quite annoying if I want to unselect multiple sensors as I have to re iterate through the menu each time I select a type of sensor to hide or show.

Acceptance Criteria:
Prevent Tool Strip Menu from closing when an end user selects any item under the hardware sub menu dropdown

Steps Took:
Added an event handler to the "menuItem5" .DropDown.Closing Event  which the object was renamed to "menuItemFileHardware" to give a more appropriate name. This event handler is to listen for the Closing Event. On Closing it shall find the cursor on the screen and see if it resides in the boundaries of the drop down. If it does, it cancel's the event

Notes On Potential Improvements:
I have not tested this with Scaling yet. To see if a user has Scaling (zoom) turned on to 125,150 if the C# winforms control will handle the scaling automatically. It might be necessary to implement a Graphics Control and take the GraphicsControl.DpiX divide it by 96 and use the floated value to multiply against X and Y. But the way it's being done right now it uses relative to click and size of display rectangle so it should be pretty safe.

### CODE REVIEW

File: LibreHardwareMonitor/UI/MainForm.Designer.cs [hunk 1]
```diff
@@ -139,7 +139,6 @@ private void InitializeComponent()
             this.splitContainer = new LibreHardwareMonitor.UI.SplitContainerAdv();
             this.treeView = new Aga.Controls.Tree.TreeViewAdv();
             this.batteryMenuItem = new System.Windows.Forms.ToolStripMenuItem();
-            this.psuMenuItem = new System.Windows.Forms.ToolStripMenuItem();
```
Comment (xmikedanielsx) [inline LibreHardwareMonitor/UI/MainForm.Designer.cs:142]: Removed this line as it was being double declared and preventing Designer from loading the Form.

**Comment by hcabel** [comment]
**Tips:** using an # and the issue number will auto assign this PL to the issue and make tracking so much easier. This can be done in commits, branch name, PL, ..

#1393

**Comment by xmikedanielsx** [comment]
> **Tips:** using an # and the issue number will auto assign this PL to the issue and make tracking so much easier. This can be done in commits, branch name, PL, ..
> 
> #1393

Roger. I forgot. Sorry.

**Comment by pedrosimoes79** [comment]
I think I have a more elegant solution, it just needs this change:
```
menuItem5.DropDown.Closing += DropDown_Closing; // on the MainForm constructor, then

private void DropDown_Closing(object sender, ToolStripDropDownClosingEventArgs e)
{
    if (e.CloseReason == ToolStripDropDownCloseReason.ItemClicked)
    {
        e.Cancel = true;
    }
}
```
Additionally, the naming should also be fixed like you already did in this PR
I was about to make my PR when I found yours LOL

**Comment by xmikedanielsx** [comment]
@pedrosimoes79 
see here the answer to the guys question. There was an old article he linked on MS forum but it doesn't work no more.
https://stackoverflow.com/questions/6826183/prevent-toolstripmenuitem-context-menu-from-closing-on-click

**Comment by xmikedanielsx** [comment]
Oh wait, I see you are doing it on the manually created item not the dynamic item. I can't remember if I tried that. I can try it again. But yes much more preferred. Will check and report back

**Comment by pedrosimoes79** [comment]
yes, it is done on the parent

**Comment by xmikedanielsx** [comment]
@pedrosimoes79  .. Implemented your solution. Much cleaner and preferred. It worked, thank you sir. I was thinking too granular.

**Comment by pedrosimoes79** [comment]
awesome, I am glad to help :)

**Pull request merged**